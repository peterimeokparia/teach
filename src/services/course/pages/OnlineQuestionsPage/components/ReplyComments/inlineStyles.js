export const replyIconStyle = (   ) => {
    return {
      fontSize: 55,
      "zIndex": 1, 
      "margin-left": "17%",
      "margin-right": "12%",
      top: "-3%",
    };
  };

  export const saveIconStyle = (   ) => {
    return {
      fontSize: 55,
      "zIndex": 1, 
      "margin-left": "17%",
      "margin-right": "12%",
      top: "-3%",
    };
  };
  
  export const deleteIconStyle = (   ) => {
    return {
      fontSize: 55,
      "zIndex": 1, 
      "margin-right": "10%",
      "margin-left": "15%",
      top: "-3%",
    };
  };

  export const iconStyle = (   ) => {
    return {
      fontSize: 60,
      "margin-left": "30px"
    };
};

export const styleObj = ( commentColor ) => {
    return {
      fontSize: 17,
      color: "black",
      textAlign: "center",
      paddingTop: "20px",
      paddingBottom: "20px",
      "background-color": commentColor,
      "border-radius": "25px"
    };
};