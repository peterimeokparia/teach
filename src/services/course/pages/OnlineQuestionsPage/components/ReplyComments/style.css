.parentCommentHighLight {
    /* background-color: whitesmoke; */
   background-color: whitesmoke;
   margin-left: 20px;
   margin-bottom: 10px;
   border-style: solid;
   border-left-width: thin;
   border-bottom-style: none;
   border-right-style: none;
   border-top-style: none;
}
.noParentCommentHighLight{
  background-color: blue;
}
.childCommentHighLight {
    margin-left: 10px;
}
.parentCommentEditorDisplay {
    font-size: 2px;
    /* border: */
}
.commentContentTop {
    margin-bottom: 30px;
}
.commentCards {
     margin-top: 30px; 
}
.comment-round-button-1 { 
  background: orange;
}
.comment-round-button-2 { 
  background: yellowgreen
}
.comment-round-button-3 { 
    background: red
}
.comment-round-button-4 { 
  background: skyblue
}
.comment-round-button-5 { 
  background: rgb(162, 118, 235);
}
.comment-round-button-6 { 
  background: white;
}
.comment-round-button-7 { 
  background: #ff5722;
}
.comment-round-button-8 { 
  background: #1976d2;
}
 .comment-round-button-1, .comment-round-button-2, 
 .comment-round-button-3, .comment-round-button-4, 
 .comment-round-button-5, .comment-round-button-6, 
 .comment-round-button-7, .comment-round-button-8 {
   align-self: flex-end; 
   width:70px;
   height:60px;
   line-height: 60px;
   border: 2px solid #f5f5f5;
   border-radius: 50%; 
   margin-top: 30px;
   margin-right: 60px;
   transition: transform .7s ease-in-out;
   box-shadow: 10px 10px black;
}
.comment-round-button-1:hover
 {
  background-image: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  background: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  cursor: pointer;
}
.comment-round-button-2:hover
 {
  background-image: linear-gradient(to left, violet, red, white, green, yellow, orange, red);
  background: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  cursor: pointer;
}
.comment-round-button-3:hover
 {
  background-image: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  background: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  cursor: pointer;
}
.comment-round-button-4:hover
 {
  background-image: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  background: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  cursor: pointer;
}
.comment-round-button-5:hover
 {
  background-image: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  background: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  cursor: pointer;
}
.comment-round-button-6:hover
 {
  background-image: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  background: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  cursor: pointer;
}
.comment-round-button-7:hover
 {
  background-image: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  background: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  cursor: pointer;
}
.comment-round-button-8:hover
 {
  background-image: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  background: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  cursor: pointer;
}
.hideDiv{
  visibility: hidden;
}
.showDiv{
  visibility: visible;
}

.collapsible {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.active, .collapsible:hover {
  background-color: #ccc;
}
.parentCommentHighLight .panel1d-header{
  background-color: white;
  border-style: none;
  border-left-width: none;
  border-bottom-style: none;
  border-right-style: none;
  border-top-style: none;
  border-color: white;
}