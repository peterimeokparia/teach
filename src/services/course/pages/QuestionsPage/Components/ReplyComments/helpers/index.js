import { 
getSortedRecordsByDate } from 'services/course/selectors';

export const getCommentIdCollection = ( onlineComments ) => {
    let commentObjects = {}, children = [];
    
    getSortedRecordsByDate(onlineComments?.filter(_comments => _comments?.commentParentId == null   ), 'commentDateTime' )?.forEach(element => {
      children = onlineComments?.find(_comments => _comments?.commentParentId === null && _comments?._id === element?._id);
      if ( children ) {
        commentObjects[ `parent_${element?._id}` ] = { commentKey: 'parent', parentkey: element?._id, children };
      }
      children = getSortedRecordsByDate(onlineComments?.filter(_comments => _comments?.commentParentId !== null && _comments?.commentParentId === element?._id  ), 'commentDateTime');
      commentObjects[ element?._id ] = { commentKey: element?._id,  parentkey: element?._id,  children };
      // children = getSortedRecordsByDate(onlineComments?.filter(_comments => _comments?.commentParentId === element?._id  ), 'commentDateTime');
      // commentObjects[ element?._id ] = { commentKey: element?._id,  parentkey: element?._id,  children };
    });
  return commentObjects;
};