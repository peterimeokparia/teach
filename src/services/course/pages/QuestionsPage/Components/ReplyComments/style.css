.parentCommentHighLight {
    /* background-color: whitesmoke; */
   background-color: whitesmoke;
   margin-left: 20px;
   margin-bottom: 10px;
   border-style: dashed;
   border-left-width: thin;
   border-bottom-style: none;
   border-right-style: none;
   border-top-style: none;
}
.childCommentHighLight {
    margin-left: 10px;
}
.parentCommentEditorDisplay {
    font-size: 2px;
    /* border: */
}
.commentContentTop {
    margin-bottom: 30px;
}
.commentCards {
     margin-top: 30px; 
}
 .comment-round-button-1 { 
    background: orange;
    /* box-shadow: 10px 10px black; */
  }
  .comment-round-button-2 { 
    background: yellowgreen
  }
  .comment-round-button-3 { 
    background: red
 }
 .comment-round-button-4 { 
  background: skyblue
}
 .comment-round-button-1, .comment-round-button-2, .comment-round-button-3, .comment-round-button-4 {
   align-self: flex-end; 
   width:70px;
   height:60px;
   line-height: 60px;
   border: 2px solid #f5f5f5;
   border-radius: 50%; 
   margin-top: 30px;
   margin-right: 60px;
   transition: transform .7s ease-in-out;
   box-shadow: 10px 10px black;
}
.comment-round-button-1:hover
 {
  background-image: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  background: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
}
.comment-round-button-2:hover
 {
  background-image: linear-gradient(to left, violet, red, white, green, yellow, orange, red);
  background: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
}
.comment-round-button-3:hover
 {
  background-image: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  background: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
}
.comment-round-button-4:hover
 {
  background-image: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
  background: linear-gradient(to left, violet, red,  green, white, green, yellow, orange, red);
}