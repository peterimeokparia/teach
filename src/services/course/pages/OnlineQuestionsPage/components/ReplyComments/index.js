
import { 
connect } from 'react-redux';

import { 
getOperatorFromOperatorBusinessName,
getSortedRecordsByDate } from 'services/course/selectors';

import {
loadOnlineComments,
addNewOnlineComment,
deleteOnlineComment,
saveOnlineComment } from 'services/course/actions/onlinecomments'; 

import {
SET_ONLINECOMMENTS_MARKDOWN } from 'services/course/actions/onlinecomments'; 

import {
getCommentIdCollection } from './helpers';

import { 
Accordion, 
AccordionSummary
} from '@material-ui/core';

import {
handleChange,
editor_upload_url } from 'services/course/pages/OnlineQuestionsPage/helpers';

import { 
styleObj, 
replyIconStyle, 
deleteIconStyle, 
iconStyle } from './inlineStyles';

import { 
saveEditorMarkDownObjectToMw } from 'services/course/actions/editor';

import useReplyCommentsHook from 'services/course/pages/OnlineQuestionsPage/hooks/useReplyCommentsHook';
import moment from "moment";
import EditorComponent  from '../../../components/EditorComponent';
import ReplyIcon from '@material-ui/icons/Reply';
import DeleteIcon from '@material-ui/icons/Delete';
import AddCommentOutlinedIcon from '@material-ui/icons/AddCommentOutlined';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import './style.css';

const ReplyComments = ({ 
  loadOnlineComments,
  addNewOnlineComment,
  deleteOnlineComment,
  saveOnlineComment,
  questionId,
  answer,
  sortedComments,
  comment,
  addComment,
  parentComment,
  onlineCommentsCollectionTest,
  operator,
  currentUser,
  currentUsers,
  comments,
  courseId, 
  question, 
  onlineQuestionAnswerId, 
  saveEditorMarkDownObjectToMw,
  parentComments,
  getSortedCommentContent,
  commentParentId }) => {

  let commentsConfig = {
    question, 
    answer, 
    currentUser, 
    operator, 
    courseId, 
    comments
  };

  let {
    addNewComment,
    onlineComments,
    onhandleSelected
  } = useReplyCommentsHook( commentsConfig );

const getMainComment = ( comment ) => {
return <div> {
  <div className={"commentCards"}>  
    <div style={styleObj( comment?.color )}> 
    <div className={""}>
    { 
        <div>
          {<div>{`ParentId ${ comment?.commentParentId }`}</div> }
          {<div>{`CommentId ${ comment?._id }`}</div>}
          { 
            <div> 
              { <span>{`Comment by ${ currentUsers?.find(_user => _user?._id === comment?.userId)?.firstname }`}</span> }
              { <span>{` on ${  moment( comment?.commentDateTime ).local() }`} </span>}
            </div>
          }
          <span> 
              <span>
                <ReplyIcon
                  style={replyIconStyle()}
                  className="comment-round-button-1"
                  onClick={() => addNewComment(comment)}
                />
              </span>
              <span>
                <DeleteIcon 
                  style={deleteIconStyle()}
                  className="comment-round-button-3"
                  onClick={() => onhandleSelected(comment)}
                />
              </span>
          </span>
          </div>     
    }
      </div>
    </div>
  </div>  
  } 
</div>;
};

const getChildComments = ( collection ) => {
  return <>  
      { collection?.map(( element, index )  => {      
      return  [ element ].map( (comment, subindex ) => {
      return <div id={comment?._id}>
        <> 
            <div className={"commentCards"}> 
            <div style={styleObj( comment?.color )}> 
            <div className={""}>
              <EditorComponent
                  className={""}
                  id={ comment?._id }
                  name={ comment?.name } 
                  handleChange={(editor) => handleChange({ ...element, markDownContent: editor }, SET_ONLINECOMMENTS_MARKDOWN, `/onlinecomments/`, saveEditorMarkDownObjectToMw )}
                  content={ element?.markDownContent }
                  upload_url={ editor_upload_url }
              /> 
              </div>
              <div>
                  { <div>{`ParentId ${ comment?.commentParentId }`}</div> }
                  { <div>{`CommentId ${ comment?._id }`}</div>  }
                  { <span>{`Comment by ${ currentUsers?.find(_user => _user?._id === comment?.userId)?.firstname }`}</span> }
                  { <span>{` on ${  moment( comment?.commentDateTime ).local() }`} </span>  }
              </div>
              <span> 
                  <span>
                    <ReplyIcon
                        style={replyIconStyle()}
                        className="comment-round-button-1"
                        onClick={() => addNewComment(comment)}
                    />
                  </span>
                  <span>
                      <DeleteIcon 
                        style={deleteIconStyle()}
                        className="comment-round-button-3"
                        onClick={() => onhandleSelected(comment)}
                      />
                  </span>
              </span> 
            </div>
            </div>
        </> 
          <div className={ "parentCommentHighLight"}> 
            { ( onlineComments?.filter( _comment => _comment?.commentParentId ===  comment?._id)?.length > 0) &&       
                <Accordion square>  
                <AccordionSummary  
                  id="panel1d-header"
                  className="panel1d-header"
                  expandIcon={ ( onlineComments?.filter( _comment => _comment?.commentParentId ===  comment?._id)?.length > 0) && <ExpandMoreIcon fontSize={'large'}/>}
                  aria-controls="panel1a-content"
                > 
                </AccordionSummary>
                {
                  <div>
                    {
                      getChildComments( getSortedRecordsByDate(onlineComments?.filter( _comment => _comment?.commentParentId ===  comment?._id), 'commentDateTime') )
                    }
                  </div>
                }
              </Accordion>       
            }     
            </div> 
      </div>;
      });
  })
  }
  </>;
};

const callOnLineComments = ( onlineComments ) => { 
  return Object.values( getCommentIdCollection( onlineComments ) )?.filter(test => test?.parentkey !== null )?.map( (answerComments, index) => {  
      if ( answerComments?.commentKey.split('_')[0] === 'parent' ) {               
            return getMainComment( answerComments?.children );
          } else {
            return<Accordion square >  
            <AccordionSummary 
              expandIcon={ ( answerComments?.children)?.length > 0 && <ExpandMoreIcon fontSize={'large'}/>}
            >
            </AccordionSummary>
            <div className={"parentCommentHighLight"}>
              {  
                getChildComments( answerComments?.children )
              }
            </div>
          </Accordion>; 
      }
  });   
};

return (
  <>
    {
      <span>
        <AddCommentOutlinedIcon
            style={iconStyle()}
            className="comment-round-button-1"
            onClick={() => addNewComment( comment )}
        />
      </span>
    }            
    { 
      callOnLineComments( onlineComments ) 
    }                         
  </>
); };

const mapState = ( state, ownProps ) => {
  return {
    operator: getOperatorFromOperatorBusinessName(state, ownProps),
    currentUser: state.users.user,
    currentUsers: Object.values( state.users.users ), 
    comments: Object.values( state?.onlineComments?.onlineComments )
  };
};
  
export default connect(mapState, { saveOnlineComment, addNewOnlineComment, loadOnlineComments, deleteOnlineComment, saveEditorMarkDownObjectToMw })(ReplyComments);
