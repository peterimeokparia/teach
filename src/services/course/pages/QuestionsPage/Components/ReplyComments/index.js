import { 
useState,
useEffect } from 'react';

import { 
connect } from 'react-redux';

import { 
getOperatorFromOperatorBusinessName,
getSortedRecordsByDate } from 'Services/course/Selectors';

import {
loadOnlineComments,
addNewOnlineComment,
deleteOnlineComment,
saveOnlineComment } from 'Services/course/Actions/OnlineComments'; 

import {
SET_ONLINECOMMENTS_MARKDOWN } from 'Services/course/Actions/OnlineComments'; 

import {
manageCommentsFieldCollection } from 'Services/course/Pages/QuestionsPage/helpers';

import {
getCalendarColor } from 'Services/course/Pages/CalendarPage/helpers';

import {
getCommentIdCollection } from './helpers';

import { 
Accordion, 
AccordionSummary
} from '@material-ui/core';

import { 
styleObj, 
replyIconStyle, 
deleteIconStyle, 
iconStyle } from './inlineStyles';

import moment from "moment";
import EditorComponent  from '../../../Components/EditorComponent';
import ReplyIcon from '@material-ui/icons/Reply';
import DeleteIcon from '@material-ui/icons/Delete';
import AddCommentOutlinedIcon from '@material-ui/icons/AddCommentOutlined';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import './style.css';

const ReplyComments = ({ 
  answer,
  sortedComments,
  loadOnlineComments,
  comment,
  addComment,
  parentComment,
  onlineCommentsCollectionTest,
  operator,
  currentUser,
  currentUsers,
  comments,
  courseId, 
  question, 
  onlineQuestionAnswerId, 
  addNewOnlineComment,
  deleteOnlineComment,
  saveOnlineComment,
  parentComments,
  getSortedCommentContent,
  commentParentId }) => {
  const [ contentChanged, setContentChanged ] = useState( false );
  let inputFieldOptions;

  useEffect(() => {
    loadOnlineComments();
    if ( contentChanged  ) {
      loadOnlineComments();
      setContentChanged( false );
    }
  }, [ loadOnlineComments, contentChanged ] );      

  let onlineComments = getSortedRecordsByDate( comments?.filter( comment => comment?.onlineQuestionId === question?._id && comment?.onlineQuestionAnswerId === answer?._id ), 'commentDateTime');

  if ( ! onlineComments ) { return <div>{''}</div>; }

const addNewComment = ( parentComment ) => {
  let config = { 
      onlineQuestionId: question?._id,
      onlineQuestionAnswerId: answer?._id,
      commentParentId: ( parentComment?._id === undefined ) ? null :  parentComment?._id,
      childComments: [],
      courseId, 
      userId: currentUser?._id, 
      placeHolder: null,
      questionCreatedBy: ( currentUser?._id ) ? (currentUser?.firstname) : 'anonymous', 
      operator: operator?._id,
      color: ( parentComment?._id === undefined ) ? getCalendarColor( parentComment ) :  parentComment?.color, 
      inputFieldOptions, 
  };

  addNewOnlineComment( manageCommentsFieldCollection(config) );
  setContentChanged( true );
};

const onhandleSelected = ( selectedComment ) => {
  deleteOnlineComment( selectedComment );
  setContentChanged( true );
};

const getMainComment = ( comment ) => {
return <div> {
  <div className={"commentCards"}>  
    <div style={styleObj( comment?.color )}> 
    <div className={""}>
    { 
        <div>
          {<div>{`ParentId ${ comment?.commentParentId }`}</div> }
          {<div>{`CommentId ${ comment?._id }`}</div>}
          { 
            <div> 
                { <span>{`Comment by ${ currentUsers?.find(_user => _user?._id === comment?.userId)?.firstname }`}</span> }
                { <span>{` on ${  moment( comment?.commentDateTime ).local() }`} </span>}
            </div>
          }
          <span> 
              <span>
                <ReplyIcon
                  style={replyIconStyle()}
                  className="comment-round-button-1"
                  onClick={() => addNewComment(comment)}
                />
              </span>
              <span>
                <DeleteIcon 
                  style={deleteIconStyle()}
                  className="comment-round-button-3"
                  onClick={() => onhandleSelected(comment)}
                />
              </span>
          </span>
          </div>     
    }
     </div>
    </div>
  </div>  
  } 
</div>;
};

const getChildComments = ( collection ) => {
  return <>  
      { collection?.map(( element, index )  => {      
      return  [ element ].map( (comment, subindex ) => {
      return <div id={comment?._id}>
        <> 
            <div className={"commentCards"}> 
            <div style={styleObj( comment?.color )}> 
            <div className={""}>
              <EditorComponent
                  className={""}
                  id={ comment?._id }
                  name={ comment?.name } 
                  editorConfiguration={{
                    enableAutoSave: true,
                    entity: comment, 
                    stateObjectType: { propNameOne: "onlineComments",  propNameTwo: "onlineComments" }, 
                    actionDescription: SET_ONLINECOMMENTS_MARKDOWN, 
                    actionObject: saveOnlineComment, 
                    duration: 2000 
                   }}
                  content={ comment?.markDownContent }
              /> 
              </div>
              <div>
                  { <div>{`ParentId ${ comment?.commentParentId }`}</div> }
                  { <div>{`CommentId ${ comment?._id }`}</div>  }
                  { <span>{`Comment by ${ currentUsers?.find(_user => _user?._id === comment?.userId)?.firstname }`}</span> }
                  { <span>{` on ${  moment( comment?.commentDateTime ).local() }`} </span>  }
              </div>
              <span> 
                  <span>
                    <ReplyIcon
                        style={replyIconStyle()}
                        className="comment-round-button-1"
                        onClick={() => addNewComment(comment)}
                    />
                  </span>
                  <span>
                      <DeleteIcon 
                        style={deleteIconStyle()}
                        className="comment-round-button-3"
                        onClick={() => onhandleSelected(comment)}
                      />
                  </span>
              </span> 
            </div>
            </div>
        </> 
          <div className={ "parentCommentHighLight"}> 
            { ( onlineComments?.filter( _comment => _comment?.commentParentId ===  comment?._id)?.length > 0) &&       
               <Accordion square>  
                <AccordionSummary  
                  id="panel1d-header"
                  className="panel1d-header"
                  expandIcon={ ( onlineComments?.filter( _comment => _comment?.commentParentId ===  comment?._id)?.length > 0) && <ExpandMoreIcon fontSize={'large'}/>}
                  aria-controls="panel1a-content"
                > 
                </AccordionSummary>
                {
                  <div>
                    {
                      getChildComments( getSortedRecordsByDate(onlineComments?.filter( _comment => _comment?.commentParentId ===  comment?._id), 'commentDateTime') )
                    }
                  </div>
                }
              </Accordion>       
            }     
            </div> 
      </div>;
      });
  })
  }
  </>;
};

const callOnLineComments = ( ) => { 
  return Object.values( getCommentIdCollection() )?.filter(test => test?.parentkey !== null )?.map( (answerComments, index) => {  
      if ( answerComments?.commentKey === 'parent' ) {               
            return getMainComment( answerComments?.children );
          } else {
            return<Accordion square >  
            <AccordionSummary 
              expandIcon={ ( answerComments?.children)?.length > 0 && <ExpandMoreIcon fontSize={'large'}/>}
            >
            </AccordionSummary>
            <div className={"parentCommentHighLight"}>
              {  
                getChildComments( answerComments?.children )
              }
            </div>
         </Accordion>; 
      }
  });   
};

return (
  <>
    {
      <span>
        <AddCommentOutlinedIcon
            style={iconStyle()}
            className="comment-round-button-1"
            onClick={() => addNewComment( comment )}
        />
      </span>
    }            
    { 
      callOnLineComments() 
    }                         
  </>
); };

const mapState = ( state, ownProps ) => {
  return {
    operator: getOperatorFromOperatorBusinessName(state, ownProps),
    currentUser: state.users.user,
    currentUsers: Object.values( state.users.users ), 
    comments: Object.values( state?.onlineComments?.onlineComments )
  };
};
  
export default connect(mapState, { loadOnlineComments, addNewOnlineComment, saveOnlineComment, deleteOnlineComment })(ReplyComments);
