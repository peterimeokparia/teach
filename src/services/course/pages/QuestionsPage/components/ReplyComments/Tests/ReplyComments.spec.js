import React from 'services/course/pages/QuestionsPage/components/ReplyComments/Tests/node_modules/react';

import { 
Provider, 
useSelector } from "services/course/pages/QuestionsPage/components/ReplyComments/Tests/node_modules/react-redux";

import { 
applyMiddleware } from 'services/course/pages/QuestionsPage/components/ReplyComments/Tests/node_modules/redux'; 

import { 
mockStoreObject } from 'services/course/pages/QuestionsPage/components/ReplyComments/Tests/node_modules/Services/course/Api';

import ReplyComments  from 'services/course/pages/QuestionsPage/components/ReplyComments';
import renderer from 'services/course/pages/QuestionsPage/components/ReplyComments/Tests/node_modules/react-test-renderer';
import thunk from "services/course/pages/QuestionsPage/components/ReplyComments/Tests/node_modules/redux-thunk"; 
import * as reactRedux from 'services/course/pages/QuestionsPage/components/ReplyComments/Tests/node_modules/react-redux';
import configureStore from 'services/course/pages/QuestionsPage/components/ReplyComments/Tests/node_modules/redux-mock-store';


  const mockStore = configureStore([]);
  
describe('ReplyComments', () =>  {  

  let  store = mockStore(mockStoreObject, applyMiddleware(thunk));

  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
 
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
    jest.spyOn(Date.prototype, 'toLocaleTimeString')
      .mockImplementation(() => '12:00:00');
    jest.spyOn(Date.prototype, 'toLocaleDateString')
      .mockImplementation(() => 'AM 1/01/2021');
  });

  it('renders as expected', async () => {

    const dispatch = jest.fn();
    useDispatchMock.mockReturnValue(dispatch)

    const tree = renderer.create(
      <Provider store={store}>
        <ReplyComments /> 
    </Provider>
    );
      expect(tree.toJSON()).toMatchSnapshot();
  })
});

