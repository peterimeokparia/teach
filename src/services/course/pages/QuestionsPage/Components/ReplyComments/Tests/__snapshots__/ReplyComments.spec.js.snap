// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReplyComments renders as expected 1`] = `
<span>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root comment-round-button-1"
    focusable="false"
    onClick={[Function]}
    style={
      Object {
        "fontSize": 60,
        "margin-left": "30px",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M22 4c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4V4zm-2 13.17L18.83 16H4V4h16v13.17zM13 5h-2v4H7v2h4v4h2v-4h4V9h-4z"
    />
  </svg>
</span>
`;
